cunumeric/deferred.py

```diff
@@ -437,20 +451,40 @@ def _create_indexing_array(self, key, is_set=False):
                 N = rhs.ndim
                 out_dtype = rhs.runtime.get_point_type(N)
 
-            out = rhs.runtime.create_unbound_thunk(out_dtype)
+            # TODO : current implementation of the ND output regions
+            # requires out.ndim == rhs.ndim. This will be fixed in the
+            # future
+            out = rhs.runtime.create_unbound_thunk(out_dtype, ndim=rhs.ndim)
+
             task = rhs.context.create_task(CuNumericOpCode.ADVANCED_INDEXING)
             task.add_output(out.base)
             task.add_input(rhs.base)
-            task.add_input(key.base)
+            task.add_input(key)
             task.add_scalar_arg(is_set, bool)
-            task.add_alignment(rhs.base, key.base)
-            task.add_broadcast(
-                key.base, axes=tuple(range(1, len(key.base.shape)))
-            )
+            task.add_scalar_arg(key_dims, ty.int64)
+            task.add_alignment(rhs.base, key)
+            task.add_broadcast(key, axes=tuple(range(1, len(key.shape))))
             task.add_broadcast(
                 rhs.base, axes=tuple(range(1, len(rhs.base.shape)))
             )
             task.execute()
+
+            # TODO : current implementation of the ND output regions
+            # requires out.ndim == rhs.ndim.
+            # The logic below will be removed in the future
+            if out_dim != rhs.ndim:
+                out_tmp = out.base
+                for dim in range(rhs.ndim - out_dim):
+                    out_tmp = out_tmp.project(rhs.ndim - dim - 1, 0)
+
+                out = self.runtime.create_empty_thunk(
+                    out_tmp.shape,
+                    out_dtype,
+                    inputs=[],
+                )
+
+                out, storet = out._copy_store(out_tmp)
```

Comment:
if you're not using the second return value, please use `_` instead. also, I don't like the way `_copy_store` is defined. what's the point of returning two values when one of them can be retrieved from the other? if that function returned only a store, then this line would have been cleaner.